assets/js/util.js

```diff
@@ -187,3 +187,16 @@ function isSubset(subset, superset) {
         return superset.indexOf(val) >= 0;
     });
 }
+
+String.prototype.hashCode = function () {
+    var hash = 0;
+    if(this.length === 0) { 
+    return hash;
+    }
+    for(i = 0; i < this.length; i++) {
+        char = this.charCodeAt(i);
+        hash = ((hash << 5) - hash) + char;
```

Comment:
I suppose, this part was written by @svineet. 
Unfortunately, I can't give a proper name.